yarn-project/txe/src/index.ts

```diff
@@ -38,6 +47,36 @@ class TXEDispatcher {
         this.logger.info(`Called reset on session ${sessionId}, yeeting it out of existence`);
       return toForeignCallResult([]);
     } else {
+      if (functionName === 'deploy') {
```

Comment:
I'd consider making a function out of this, and perhaps including sample Noir syntax in the comments

Reply:
Refactored!